app/operations/fdsh/jobs/generate_transmittable_ssa_payload.rb

```diff
@@ -0,0 +1,139 @@
+
+          return Success(ssa_person) if ssa_person.persisted?
+          add_errors({ job: @job, transmission: @transmission }, "Unable to save person subject due to #{ssa_person.errors&.full_messages}",
+                     :create_ssa_subject)
+          status_result = update_status({ job: @job, transmission: @transmission }, :failed, "Unable to save person subject")
+          return status_result if status_result.failure?
+          Failure("Unable to save person subject")
+        end
+      end
+
+      def create_transaction(values, subject)
+        result = Fdsh::Jobs::CreateTransaction.new.call(values.merge({ transmission: @transmission,
+                                                                       subject: subject,
+                                                                       event: 'initial',
+                                                                       state_key: :initial }))
+        return result if result.success?
+        add_errors({ job: @job, transmission: @transmission }, "Failed to create transaction due to #{result.failure}", :create_transaction)
+        status_result = update_status({ job: @job, transmission: @transmission, transaction: @transaction }, :failed, result.failure)
+        return status_result if status_result.failure?
+        result
+      end
+
+      def generate_transmittable_payload(payload)
+        result = Fdsh::Ssa::H3::TransformPersonToJsonSsa.new.call(payload)
+        if result.success?
+          @transaction.json_payload = result.value! if result.success?
+          @transaction.save
+
+          return Success(@transaction) if @transaction.json_payload
+          add_errors({ job: @job, transmission: @transmission, transaction: @transaction },
+                     "Unable to save transaction with payload",
+                     :generate_transmittable_payload)
+          status_result = update_status({ job: @job, transmission: @transmission, transaction: @transaction }, :failed,
+                                        "Unable to save transaction with payload")
+          return status_result if status_result.failure?
+        else
+          add_errors({ job: @job, transmission: @transmission, transaction: @transaction },
+                     "Unable to transform payload",
+                     :generate_transmittable_payload)
+          status_result = update_status({ job: @job, transmission: @transmission, transaction: @transaction }, :failed, "Unable to transform payload")
+          return status_result if status_result.failure?
+          result
+        end
+      end
+
+      def transmittable
+        message_id = @job.message_id
+
+        if @transaction.json_payload && message_id
+          Success({ transaction: @transaction,
+                    message_id: message_id })
+        else
+          add_errors({ job: @job, transmission: @transmission, transaction: @transaction },
```

Comment:
not: can we store this message into a variable `"Transaction do not consists of a payload or no message id found"` and reuse so it would be easier if this needs to be updated for some reason. 

Similar things need to happen in other cases here.

Reply:
If you look into the add_errors method we actually have transmittable_error class. That we save the errors. This is not `errors.add` instead it is a method.